core/domain/user_query_services.py

```diff
@@ -35,14 +36,11 @@ def _get_user_query_from_model(user_query_model):
     Returns:
         UserQuery. User query domain object.
     """
-    user_query_params = user_query_domain.UserQueryParams(
-        user_query_model.inactive_in_last_n_days,
-        user_query_model.has_not_logged_in_for_n_days,
-        user_query_model.created_at_least_n_exps,
-        user_query_model.created_fewer_than_n_exps,
-        user_query_model.edited_at_least_n_exps,
-        user_query_model.edited_fewer_than_n_exps
-    )
+    attributes = [
+        getattr(user_query_model, predicate['backend_attr']) for predicate in (
+            constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION)]
+    user_query_params = user_query_domain.USER_QUERY_PARAMS(*attributes)
+
```

Comment:
Doesn't this make us dependent on the order of the `EMAIL_DASHBOARD_PREDICATE_DEFINITION`? Should we rather use kwargs so that the order of the attributes doesn't matter?

Reply:
I think the order does not matter for a named tuple, here in the current implementation the order is not considered.

Reply:
I have changed it to a dict which does not consider the order.